src/containers/assignments/AssignmentFormContainer.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+
+import AssignmentForm from '../../components/assignments/AssignmentForm';
+import {
+  ASSIGNMENTS_INITIAL_STATE, ASSIGNMENTS_PROPTYPES
+} from '../../ducks/assignments';
+import {
+  CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
+} from '../../ducks/classrooms';
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
+
+export class AssignmentFormContainer extends React.Component {
+  constructor() {
+    super();
+
+    this.createAssignment = this.createAssignment.bind(this);
+    this.updateAssignment = this.updateAssignment.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onChangeDate = this.onChangeDate.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillUnmount() {
+    this.resetFormFields();
+  }
+
+  onChange(event) {
+    if (event.target && event.target.id && event.target.value) {
+      const fields = { ...this.props.formFields, [event.target.id]: event.target.value };
+      Actions.assignments.updateFormFields(fields);
+    }
+  }
+
+  onChangeDate(datetime) {
+    // Grommet's DateTime passes the value directly rather than the standard event object
+    // TODO: why?
+    const fields = { ...this.props.formFields, duedate: datetime };
+    Actions.assignments.updateFormFields(fields);
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+
+    if (this.props.selectedAssignment) {
+      this.updateAssignment().then(() => {
+        if (this.props.onSubmit) this.props.onSubmit();
+      });
+    } else {
+      this.createAssignment().then(() => {
+        if (this.props.onSubmit) this.props.onSubmit();
+      });
+    }
+  }
+
+  createAssignment() {
+    const assignmentData = {
+      attributes: {
+        name: this.props.formFields.name,
+        metadata: {
+          classification_target: this.props.formFields.classification_target,
```

Comment:
It seems this is using `classification_target` instead of the plural form `classification_target`; I'm assuming that the plural form is the correct one.

Side note: I noticed the non-plural spelling on `ClassroomFormContainer.jsx` as well.

Reply:
I made the singular v. plural mistake all over the place and I don't think I caught until after I started working on this. It might be the source of the 422?